src/storage/mvcc/txn.rs

```diff
@@ -419,8 +422,8 @@ impl<S: Snapshot> MvccTxn<S> {
         min_commit_ts: TimeStamp,
     ) -> Result<()> {
         let lock_type = LockType::from_mutation(&mutation);
-        // For the insert operation, the old key should not be in the system.
-        let should_not_exist = mutation.is_insert();
+        // For the insert/checkNotExists operation, the old key should not be in the system.
+        let should_not_exist = mutation.should_not_exists();
```

Comment:
I prefer to remove the `Noop` type lock. Just check key-not-exist if the type of mutation is `CheckNotExists`.